### PR TITLE
4.1 release

### PR DESCRIPTION
**Pull request by apaparazzi0329**
The moment we've all been waiting for. Swift Package Manager support for all services and XCode 12 compatibility

### CODE REVIEW

**Comment by repjarms** [comment]
@apaparazzi0329 we need to update `.travis.yml` to run all of the jobs on `XCode 12`.

File: README.md [hunk 1]
```diff
@@ -79,7 +78,55 @@ IBM Watson™ Compare and Comply is discontinued. Existing instances are support
...
+The IBM Watson Swift SDK can be installed with [Swift Package Manager](https://swift.org/package-manager/), [Cocoapods](http://cocoapods.org/), or [Carthage](https://github.com/Carthage/Carthage).
+
+### Swift Package Manager
+New in version 4.1, the Watson Developer Cloud Swift SDK now supports all services through Swift Package Manager.
+
+On the XCode menu bar at the top of your screen click `File -> Swift Packages -> Add Package Dependencies`, follow the prompts by pasting the github url `https://github.com/watson-developer-cloud/swift-sdk` and using the most recent major version if appropriate. Make sure to only click on the services you plan on using otherwise you may face long build times
+
+To import a service into your project:
+```swift
+import AssitantV2
```
Comment (repjarms) [inline README.md:90]:
```suggestion
import AssistantV2
```

File: README.md [hunk 2]
```diff
@@ -79,7 +78,55 @@ IBM Watson™ Compare and Comply is discontinued. Existing instances are support
...
+import DiscoveryV2
+.
+.
+.
+```
+
+
+
+(**Speech To Text and Text To Speech only**)
```
Comment (repjarms) [inline README.md:99]:
```suggestion
**(Speech To Text and Text To Speech only)**
```

File: README.md [hunk 3]
```diff
@@ -79,7 +78,55 @@ IBM Watson™ Compare and Comply is discontinued. Existing instances are support
...
+
+The use of special libraries by these services requires extra steps.
```
Comment (repjarms) [inline README.md:101]:
```suggestion
The use of the `libogg` and `opus` libraries by these services requires extra steps to be taken.
```

File: README.md [hunk 4]
```diff
@@ -95,18 +142,18 @@ If your project does not yet have a Podfile, use the `pod init` command in the r
...
-    pod 'IBMWatsonDiscoveryV1', '~> 4.0.1'
-    pod 'IBMWatsonLanguageTranslatorV3', '~> 4.0.1'
-    pod 'IBMWatsonNaturalLanguageClassifierV1', '~> 4.0.1'
-    pod 'IBMWatsonNaturalLanguageUnderstandingV1', '~> 4.0.1'
-    pod 'IBMWatsonPersonalityInsightsV3', '~> 4.0.1'
-    pod 'IBMWatsonSpeechToTextV1', '~> 4.0.1'
-    pod 'IBMWatsonTextToSpeechV1', '~> 4.0.1'
-    pod 'IBMWatsonToneAnalyzerV3', '~> 4.0.1'
-    pod 'IBMWatsonVisualRecognitionV3', '~> 4.0.1'
+    pod 'IBMWatsonAssistantV1', '~> 4.0'
```
Comment (repjarms) [inline README.md:145]: These version numbers are updated by the release automation and shouldn't be hand edited.

File: README.md [hunk 5]
```diff
@@ -131,38 +180,57 @@ $ brew install carthage
 If your project does not have a Cartfile yet, use the `touch Cartfile` command in the root directory of your project. To install the IBM Watson Swift SDK using Carthage, add the following to your Cartfile.
 
 ```
-github "watson-developer-cloud/swift-sdk" ~> 4.0.1
+github "watson-developer-cloud/swift-sdk" ~> 4.0
```
Comment (repjarms) [inline README.md:183]: Same as with the pod installation instructions above, this version number shouldn't be hand edited.

File: README.md [hunk 6]
```diff
@@ -79,7 +78,55 @@ IBM Watson™ Compare and Comply is discontinued. Existing instances are support
 
 ## Installation
 
-The IBM Watson Swift SDK can be installed with [Cocoapods](http://cocoapods.org/), [Carthage](https://github.com/Carthage/Carthage), or [Swift Package Manager](https://swift.org/package-manager/).
+The IBM Watson Swift SDK can be installed with [Swift Package Manager](https://swift.org/package-manager/), [Cocoapods](http://cocoapods.org/), or [Carthage](https://github.com/Carthage/Carthage).
+
+### Swift Package Manager
+New in version 4.1, the Watson Developer Cloud Swift SDK now supports all services through Swift Package Manager.
```
Comment (repjarms) [inline README.md:84]: This references version `4.1`, but so far the highest commit level is a `fix`. This means that the semantic release is likely to resolve to version `4.0.2`.

**Review by repjarms — DISMISSED** [review]
This is really great work. I can't wait to get this merged in and resolve some of the package management issues we've been having.

There are some changes needed before we can merge. I've detailed them here.

Also, as I noted in my other comment, we should update `.travis.yml` to run all of the build phases on XCode 12.

File: README.md [hunk 7]
```diff
@@ -79,7 +78,55 @@ IBM Watson™ Compare and Comply is discontinued. Existing instances are support
...
+
+5. Replace the currently installed `libogg` and `libopus` libraries
+    ```bash
+    rm /usr/local/Cellar/libogg/1.3.4/lib/libogg.a && cp ~/Downloads/libogg.a /usr/local/Cellar/libogg/1.3.4/lib
+    ```
+    ```bash
+    rm /usr/local/Cellar/libogg/1.3.1/lib/libopus.a && cp ~/Downloads/libopus.a /usr/local/Cellar/opus/1.3.1/lib
+    ```
+
+6. You're ready to go!
```
Comment (mediumTaj) [inline README.md:129]: 🚀

**Review by mediumTaj — APPROVED** [review]
👍 looks good! Thanks for your work on this Angelo! 1 question, can `InsecureConnection.swift` be moved to some common library or do we need one for each serice?

**Comment by apaparazzi0329** [comment]
@mediumTaj Unfortunately swift package manager compiles from a single source folder so any source that needs to be compiled for a service has to be in the same respective folder.

File: README.md (hunk 6 above)
Comment (apaparazzi0329) [reply README.md:84]: My bad I thought a fix resolved to a minor version update

**Comment by repjarms** [comment]
FYI that this fixes #1044 and #1046

**Review by mkistler — APPROVED** [review]
Looks good. 👍

**Pull request merged**